### ISSUE TITLE
Calling close does not close the socket if there is ongoing recv 

### ISSUE THREAD

**Issue by Kyry** [Gr1N/nats-python#1]
Calling `close` on socket does not close the socket if there is pending operation
This could be fixed by using shutdown method

```
        client._socket_file.shutdown(socket.SHUT_RDWR)
        client._socket.shutdown(socket.SHUT_RDWR)
        client.close()
```

**Comment by Gr1N** [comment]
Hi, thanks for reporting.

As I see it can be easily fixed by adding two lines of code to `close` method (https://github.com/Gr1N/nats-python/blob/master/pynats/client.py#L146). I will try to fix it this week, but any PRs to fix the issue from your side are welcome.

**Comment by Gr1N** [comment]
Sorry, but I don't have free time to fix it. Anyway, PR is welcome!

**Comment by richard78917** [comment]
This should be resolved by this PR:
https://github.com/Gr1N/nats-python/pull/11

**Comment by Gr1N** [comment]
Fixed in #11. @richard78917 thanks!
